staging/src/k8s.io/legacy-cloud-providers/gce/gce_disks.go

```diff
@@ -620,6 +624,37 @@ func (g *Cloud) DisksAreAttached(diskNames []string, nodeName types.NodeName) (m
 	return attached, nil
 }
 
+// BulkDisksAreAttached is a batch function to check if all corresponding disks are attached to the
+// nodes specified with nodeName.
+func (g *Cloud) BulkDisksAreAttached(diskByNodes map[types.NodeName][]string) (map[types.NodeName]map[string]bool, error) {
+	instanceNames := []string{}
+	for nodeName := range diskByNodes {
+		instanceNames = append(instanceNames, mapNodeNameToInstanceName(nodeName))
+	}
+
+	// List all instances with the given instance names
+	// Then for each instance listed, add the disks attached to that instance to a map
+	listedInstances, err := g.getFoundInstanceByNames(instanceNames)
```

Comment:
It seems we only use disk names of the return value from Cloud#getFoundInstanceByNames .
Instead of mapping to []*compute.AttachedDisk, listedInstanceNamesToDisks can map to Set of disk names which verifyDisksAttachedToNode can directly utilize ?

@jingxu97 @hantaowang what do you think ?